app/models/host/base.rb

```diff
@@ -23,6 +23,7 @@ class Base < ActiveRecord::Base
             :conditions => { :provision => true }
     has_one :domain, :through => :primary_interface
     has_one :subnet, :through => :primary_interface
+    has_many :host_statuses, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host
```

Comment:
I think this is missing a `:dependent => :destroy`, as status entries aren't being deleted on host deletion.